OrbitCore/PathTest.cpp

```diff
@@ -96,3 +96,171 @@ TEST(Path, JoinPathPartsAbsoluteRootBackslash) {
+  EXPECT_EQ(Path::StripExtension(input), "no/ext");
+  EXPECT_EQ(Path::GetFileName(input), "ext");
+  EXPECT_EQ(Path::GetFileNameNoExt(input), "ext");
+
+  input = "double.ext.txt";
+  EXPECT_EQ(Path::GetExtension(input), ".txt");
+  EXPECT_EQ(Path::StripExtension(input), "double.ext");
+  EXPECT_EQ(Path::StripExtension(Path::StripExtension(input)), "double");
+  EXPECT_EQ(Path::GetFileName(input), "double.ext.txt");
+  EXPECT_EQ(Path::GetFileNameNoExt(input), "double.ext");
+}
+
+// TODO: A test for GetExecutableDir is missing - using this here makes this test
+// kind of invalid, since GetExecutableDir is using GetExecutablePath internally...
+TEST(Path, GetExecutablePath) {
+  std::string actual = Path::GetExecutablePath();
+  std::string expected = Path::JoinPath({Path::GetExecutableDir(), "OrbitCoreTests.exe"});
+  EXPECT_EQ(actual, expected);
+}
+
+static std::string GetTestDataAbsPath(const std::string& filename = "") {
+  if (filename.empty()) {
+    return Path::JoinPath({Path::GetExecutableDir(), "testdata", "OrbitCore"});
+  } else {
+    return Path::JoinPath({Path::GetExecutableDir(), "testdata", "OrbitCore", filename});
+  }
+}
+
+TEST(Path, GetDirectoryAndParent) {
+  struct Test {
+    std::string input;
+    std::string dir;
+    std::string parent_of_input;
+    std::string parent_of_dir;
+  };
+
+  // The hoops I'm jumping through to have the test declarations below nicely formatted...
+  using V = std::vector<Test>;
+#ifdef _WIN32
+  V tests = {{"C:\\dir/With\\fwd_slash.txt.ext", "C:/dir/With/", "C:/dir/With/", "C:/dir/"},
+             {"C:\\no\\ext\\file", "C:/no/ext/", "C:/no/ext/", "C:/no/"},
+             {"C:\\dir\\subdir\\", "C:/dir/subdir/", "C:/dir/", "C:/dir/"},
+             {"C:\\file_in_root", "C:/", "C:/", ""},
+             {"broken_path", "", "", ""},
+             {"C:\\", "C:/", "", ""}};
+#else
+  V tests = {{"/dir/With/mult_ext.txt.ext", "/dir/With/", "/dir/With/", "/dir/"},
+             {"/no/ext/file", "/no/ext/", "/no/ext/", "/no/"},
+             {"/dir/subdir/", "/dir/subdir/", "/dir/", "/dir/"},
+             {"/file_in_root", "/", "/", ""},
+             {"broken_path", "", "", ""},
+             {"/", "/", "", ""}};
+#endif
+
+  for (size_t i = 0; i < tests.size(); ++i) {
+    std::cout << "Checking directory and parent of \"" << tests[i].input << "\"\n";
+    const std::string result_dir = Path::GetDirectory(tests[i].input);
+    EXPECT_EQ(result_dir, tests[i].dir);
+    EXPECT_EQ(Path::GetParentDirectory(tests[i].input), tests[i].parent_of_input);
+    EXPECT_EQ(Path::GetParentDirectory(result_dir), tests[i].parent_of_dir);
+  }
+}
+
+// TODO: This would be more meaningful with absolute paths that are
+// created explicitely for the test
+TEST(Path, DirExists) {
+  auto dir = Path::GetDirectory(Path::GetExecutableDir());
+  EXPECT_TRUE(Path::DirExists(dir));
+  EXPECT_FALSE(Path::DirExists(Path::JoinPath({dir, "__invalid_subdir__"})));
+}
+
+TEST(Path, FileSize) {
```

Comment:
Maybe add a test for FileSize of non-existing file. Currently the result is 0 in that case.

Reply:
Removed FileSize, so the test is obsolete